hostdb/hostdb.go

```diff
@@ -49,15 +49,26 @@ func ForEachAnnouncement(b types.Block, height uint64, fn func(types.PublicKey,
 			}
 
 			fn(hostKey, Announcement{
-				Index: types.ChainIndex{
-					Height: height,
-					ID:     b.ID(),
-				},
+				Index:      ci,
 				Timestamp:  b.Timestamp,
 				NetAddress: string(ha.NetAddress),
 			})
 		}
 	}
+	for _, txn := range b.V2Transactions() {
+		for _, att := range txn.Attestations {
+			if att.Key != "HostAnnouncement" {
+				continue
+			} else if len(att.Value) == 0 {
```

Comment:
is this strictly necessary? Seems like it would be handled by a later validity check anyway